# InfectiousDiseaseResource

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.4165001.svg)](https://doi.org/10.5281/zenodo.4165001)  

Infectious diseases (ID) represent a significant proportion of morbidity and mortality across the world. Host genetic variation is likely to contribute to ID risk and downstream clinical outcomes,  but there is a need for a genetics-anchored framework to decipher molecular mechanisms of disease risk, infer causal effect on potential complications, and identify instruments for drug target discovery. Here we perform transcriptome-wide association studies (TWAS) of 35 clinical ID traits in a cohort of 23,294 individuals, identifying 70 gene-level associations with 26 ID traits. Replication in two large-scale biobanks provides additional support for the identified associations. A phenome-scale scan of the 70 gene-level associations across hematologic, respiratory, cardiovascular, and neurologic traits proposes a molecular basis for known complications of the ID traits. Using Mendelian Randomization, we then provide causal support for the effect of the ID traits on adverse outcomes. The rich resource of genetic information linked to serologic tests and pathogen cultures from bronchoalveolar lavage, sputum, sinus/nasopharyngeal, tracheal, and blood samples (up to 7,699 positive pathogen cultures across 92 unique genera) and a large catalog of genome-wide associations of microbiome variation generated from phylogenetic analysis of 16S rRNA gene sequences are developed here into a platform to interrogate the genetic basis of compartment-specific infection and colonization. To accelerate insights into cellular mechanisms, we develop a TWAS repository of gene-level associations in a broad collection of human tissues with 79 pathogen-exposure induced cellular phenotypes as a discovery and replication platform. Cellular phenotypes of infection by 8 pathogens included pathogen invasion, intercellular spread, cytokine production, and pyroptosis. These rich datasets will facilitate mechanistic insights into the role of host genetic variation on ID risk and pathophysiology, with important implications for our molecular understanding of potentially severe phenotypic outcomes.

## Reference:

The genetic architecture of human infectious diseases and pathogen-induced cellular phenotypes

Andrew T. Hale<sup>1,2</sup>, Dan Zhou<sup>2</sup>, Rebecca L. Sale<sup>2</sup>, Lisa Bastarache<sup>3</sup>, Liuyang Wang<sup>4</sup>, Steven J. Schiff<sup>5</sup>, Dennis C. Ko<sup>4</sup>, and Eric R. Gamazon<sup>2,6,7*</sup>

1 Vanderbilt University School of Medicine, Medical Scientist Training Program, Nashville, TN.  
2 Vanderbilt Genetics Institute & Division of Genetic Medicine, Vanderbilt University Medical Center, Nashville, TN.  
3 Department of Bioinformatics, Vanderbilt University School of Medicine, Nashville, TN.  
4 Department of Molecular Genetics and Microbiology, Duke University School of Medicine, Durham, NC.  
5 Center for Neural Engineering, Departments of Neurosurgery, Engineering Science and Mechanics, and Physics, Penn State University. State College, PA.  
6 Clare Hall, University of Cambridge, Cambridge, UK.  
7 MRC Epidemiology Unit, University of Cambridge, Cambridge, UK  

* Correspondence should be addressed to E.R.G. (ericgamazon@gmail.com) 

## Preprint:

This repository is the project URL for the preprint on medRxiv: https://www.medrxiv.org/content/10.1101/2020.07.19.20157404v2.full 

## Code:

Software developed and maintained by ER Gamazon and D Zhou.
